### ISSUE TITLE
`disable` comment on function gets ignored when decorator is followed by comment line

### ISSUE THREAD

**Issue by jszopi** [PyCQA/pylint#3035]
### Steps to reproduce

Hi, due to a false positive reported in https://github.com/PyCQA/pylint/issues/414, a `method-hidden` exclusion is necessary in the snippet below. I found that if the function is decorated and a comment line follows the decorator, the exclusion comment gets ignored:

```py
 1  import json
 2  
 3  def decor(f):
 4      return f
 5  
 6  class MyEncoder(json.JSONEncoder):
 7      # Comment 1 - benign
 8      @decor
 9      # Comment 2 - causes the `disable` comment below to be ignored
10      def default(self, o):  # pylint: disable=method-hidden
11          json.JSONEncoder.default(self, o)
```

### Current behavior

The "Comment 2" line causes the `method-hidden` warning to be raised.

```
code.py:9:4: E0202: An attribute defined in json.encoder line 158 hides this method (method-hidden)
```

Interestingly the error is reported on the line of the comment, rather than the function definition itself.

### Expected behavior

Adding the "Comment 2" line should have no influence on the pylint check. The issue only occurs when both the decorator and the comment between the decorator and the function are present.

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.7.2 (default, Jul 14 2019, 19:42:18)
```

**Comment by PCManticore** [comment]
Thanks for the report.

**Comment by DanielNoord** [comment]
For future contributors:
This is due to pylint thinking that the definition of `default` happens on line 9 instead of line 10. I have seen other issues that also result from incorrect inference of function declarations in combination with docstrings/decorators.

**Comment by edwardfung123** [comment]
By any chance that this can be fixed?

**Comment by DanielNoord** [comment]
In the upcoming update of `astroid` we might be able to infer this a little bit better, but I can't make any promises. This is actually not that easy to solve as it is not an internal `pylint` issue, but rather an issue depending on how we build the `ast` representation of the code (or rather how the `stdlib` `ast` module does so).

**Comment by DanielNoord** [comment]
This has been fixed and can no longer be reproduced on main. See:
```python
import json


def decor(f):
    return f


class MyEncoder(json.JSONEncoder):
    # Comment 1 - benign
    @decor
    # Comment 2 - causes the `disable` comment below to be ignored
    def default(self, c):  # pylint: disable=arguments-renamed
        json.JSONEncoder.default(self, c)
```
